### PR TITLE
CMake Support for Linux

### PR DESCRIPTION
**Pull request by EggBanana123**
Changed the readme to reflect this too

### CODE REVIEW

File: platforms/Linux.cmake [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    if(NOT GLEW_FOUND)
+        message(NOTICE "could not find glew, attempting to build from source")
+
+    else()
+        message("found GLEW")
+        target_link_libraries(RetroEngine ${GLEW_STATIC_LIBRARIES})
+        target_link_options(RetroEngine PRIVATE ${GLEW_STATIC_LDLIBS_OTHER})
+        target_compile_options(RetroEngine PRIVATE ${GLEW_STATIC_CFLAGS})
+    endif()
+endif()
```
Comment (MegAmi24) [inline platforms/Linux.cmake]: v2 doesn't have a hardware renderer.

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,173 @@
+cmake_minimum_required(VERSION 3.7)
+
+project(RetroEngine)
+
+option(FORCE_CASE_INSENSITIVE "Forces case insensivity." OFF)
```
Comment (MegAmi24) [inline CMakeLists.txt:5]: This option doesn't exist in v2.

File: README.md [hunk 3]
```diff
@@ -29,6 +30,21 @@ Without assets from the Sonic Nexus 2008 demo, this decompilation will not run.
   * The `CXXFLAGS` option can be removed if you do not want optimizations.
   * -j switch is optional, but will make building faster by running it parallel on multiple cores (8 cores would be -j9.)
 
+### CMake
+* Install the following dependencies depending on your platform through your terminal:
+  * **pacman (Arch):** `sudo pacman -S base-devel cmake glew sdl2 libogg libtheora libvorbis`
+  * **apt (Debian/Ubuntu):** `sudo apt install build-essential cmake libglew-dev libglfw3-dev libsdl2-dev libogg-dev libtheora-dev libvorbis-dev`
+  * **rpm (Fedora):** `sudo dnf install make gcc cmake glew-devel glfw-devel sdl2-devel libogg-devel libtheora-devel libvorbis-devel zlib-devel`
+  * **apk (Alpine/PostmarketOS)** `sudo apk add build-base cmake glew-dev glfw-dev sdl2-dev libogg-dev libtheora-dev libvorbis-dev`
```
Comment (MegAmi24) [inline README.md]: GLEW and GLFW aren't used for v2.

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+option(RETRO_MOD_LOADER "Enables or disables the mod loader." ON)
+option(USE_HW_REN "Enables usage of Hardware Render as an option." ON)
```
Comment (MegAmi24) [inline CMakeLists.txt]: v2 doesn't have a hardware renderer.

File: CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+    target_include_directories(RetroEngine PRIVATE
+        dependencies/${DEP_PATH}/
+    )
+endif()
+
+
+target_compile_definitions(RetroEngine PRIVATE
+    RETRO_USE_MOD_LOADER=$<BOOL:${RETRO_USE_MOD_LOADER}>
+    USE_HW_REN=$<BOOL:${USE_HW_REN}>
+    FORCE_CASE_INSENSITIVE=$<BOOL:${FORCE_CASE_INSENSITIVE}>
```
Comment (MegAmi24) [inline CMakeLists.txt]: These options don't exist in v2.

File: CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+option(USE_HW_REN "Enables usage of Hardware Render as an option." ON)
+option(RETRO_USE_ORIGINAL_CODE "Removes any change that differs from the original code, a playable game can't be built this way." OFF)
+
+set(RETRO_SDL_VERSION 2 CACHE STRING "Select between SDL2 and SDL1, defaults to SDL2")
+
+if(RETRO_ORIGINAL_CODE)
+    set(RETRO_MOD_LOADER OFF)
+endif()
+
+set(RETRO_NAME "nexus")
```
Comment (MegAmi24) [inline CMakeLists.txt]: Name should be `Nexus` (capitalize the N).

File: CMakeLists.txt (hunk 2 above)
Comment (xx-kotas-xx) [reply CMakeLists.txt:5]: It does seem to, unless it's a leftover from the v3 deomp, which the v2 decomp does seem to be based off

(Reader.hpp)
image
(Makefile)
image

Comment (MegAmi24) [reply CMakeLists.txt:5]: Seems you're right, I checked Reader.cpp instead of Reader.hpp when I was checking if it existed, my bad

**Pull request merged**